src/util/addDecorationsToEditor.ts

```diff
@@ -169,3 +168,33 @@ export function addDecorationsToEditors(
     });
   });
 }
+
+interface Lexeme {
+  /** The normalised text of the lexeme. */
+  text: string;
+
+  /** The start offset of the lexeme within its containing token */
+  tokenStartOffset: number;
+
+  /** The end offset of the lexeme within its containing token */
+  tokenEndOffset: number;
+}
+
+function getTokenLexemes(
+  text: string,
+  tokenHatSplittingMode: TokenHatSplittingMode
+): Lexeme[] {
+  switch (tokenHatSplittingMode) {
+    case "standard":
+    case "preserveCase": {
+      const normalisedText =
+        tokenHatSplittingMode === "preserveCase" ? text : text.toLowerCase();
```

Comment:
Doing a fall through and then doing a check for the switch value inside case to do different behavior is a bit weird. Why not just remove the switch?

Reply:
Was trying to plan ahead.  But I think I'll just switch to boolean anyway